detail/detail.js

```diff
@@ -0,0 +1,31 @@
+import { getCat } from '../fetch-utils.js';
+
+const mainElDetailsContainer = document.querySelector('main');
+
+let catData = {};
+
+window.addEventListener('load', async () => {
+    const data = new URLSearchParams(window.location.search);
+    const id = data.get('id');
+    const cat = await getCat(id);
+
+    const div = document.createElement('div');
```

Comment:
the rubric says we need a PURE function for rendering the cat details. you could achieve this by refactoring lines 12 - 27 into a render function in your render-utils.js. then loop through your array, and for each cat, append the render function onto the DOM.